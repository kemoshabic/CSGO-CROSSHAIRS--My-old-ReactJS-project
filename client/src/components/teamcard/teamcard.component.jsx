import React from "react";
import { connect } from "react-redux";
import { addPlayersInfo } from "../../redux/teams/teams.actions";

import "./teamcard.styles.scss";

export const TeamCard = ({ section, addPlayersInfo }) => {
  const { team, imageUrl, color, players, teamColor, teamsData } = section;
  return (
    <div className="directory-item">
      <div className="directory-item__heading-box " style={{ backgroundColor: `${color}` }}>
        <h3 className={`directory-item__heading ${ teamColor ? "directory-item__heading--black" : ""}`}>
          {team}
        </h3>
      </div>
      {players.map((item, i) => (
        <div key={i} id='test-id' className={`${ i % 2 === 0 ? "directory-item__grey-box" : "directory-item__white-box" }`}
          onClick={() => addPlayersInfo({ ...teamsData[i], team, imageUrl })}
        >
          <p className="directory-item__player">{players[i]}</p>
        </div>
      ))}
      <div className="directory-item__logo" style={{ backgroundImage: `url(${imageUrl})` }} />
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  addPlayersInfo: (info) => dispatch(addPlayersInfo(info)),
});

export default connect(null, mapDispatchToProps)(TeamCard);