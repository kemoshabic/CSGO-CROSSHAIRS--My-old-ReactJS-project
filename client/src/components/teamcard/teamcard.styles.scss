.directory-item{
  width: 25rem;
  height: 43rem;
  border-top-right-radius: 2.1rem;
  background-color: #FFFFFF;

  &__heading-box{
    height: 8rem;
    border-top-right-radius: 2rem;
    background-color: #FF5900;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  &__heading{
    font-size: 2.4rem;
    &--black{
      color: #000000;

    }
  }
  &__grey-box{
    cursor: pointer;
    background-color:#EBEBEB;
    height: 5.6rem;

  }

  &__white-box{
    cursor: pointer;
    background-color:#FFFFFF;
    height: 5.6rem;
  }

  &__player{
    padding-top: 1.5rem;
    color: #000000;
    font-size: 1.8rem;
    display: flex;
    justify-content: center;
    align-items: center;
    &:hover{
      color: #6843CF;
    }
  }
  &__logo{
    height: 8.3rem;
    width: 25rem;
    background-color: #FFFFFF;
    background-repeat: no-repeat;
    background-position: center;
  }
}

