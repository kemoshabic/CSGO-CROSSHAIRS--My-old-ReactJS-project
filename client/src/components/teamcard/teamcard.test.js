import React from "react";
import { shallow } from "enzyme";
import { TeamCard } from "./teamcard.component";

describe("TeamCard component", () => {
  let wrapper;
  let mockAddPlayersInfo;
  const mockTeam = "Faze";
  const mockImageUrl = "www.image.com";
  const mockColor = "#222222";
  let mockPlayers = [];
  const mockTeamColor = "#255555";
  let mockTeamData = [];

  beforeEach(() => {

    mockAddPlayersInfo = jest.fn();

    const mockProps = {
      section: {
        team: mockTeam,
        imageUrl: mockImageUrl,
        color: mockColor,
        players: mockPlayers,
        teamColor: mockTeamColor,
        teamsData: mockTeamData,
      },
      addPlayersInfo: mockAddPlayersInfo,
    };

    wrapper = shallow(<TeamCard {...mockProps} />);
  });

  it("should render TeamCard component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should return no players because no players are provided", () => {
    expect(wrapper.find(".directory-item__grey-box").length).toEqual(0);
  });
  it("should render team prop in Team div", () => {
    expect(wrapper.find(".directory-item__heading").text()).toBe(mockTeam);
  });
  it("should call addPlayersInfo if players are provided", () => {
    mockPlayers = ["dev1ce"];
    mockTeamData = [
      {
        id: 1,
        name: "text",
        img: "text",
        role: "text",
        crosshairImg: "text",
        settings: {},
        crosshair: "text",
      },
    ];

    const mockProps = {
      section: {
        team: mockTeam,
        imageUrl: mockImageUrl,
        color: mockColor,
        players: mockPlayers,
        teamColor: mockTeamColor,
        teamsData: mockTeamData,
      },
      addPlayersInfo: mockAddPlayersInfo,
    };
    const newWrapper = shallow(<TeamCard {...mockProps} />);

    newWrapper.find("#test-id").first().simulate("click");
    expect(mockAddPlayersInfo).toHaveBeenCalled();
  });

  it("should render first players name", () => {
    expect(wrapper.find(".directory-item__player").first().text()).toBe(mockPlayers[0]);
  });
});
