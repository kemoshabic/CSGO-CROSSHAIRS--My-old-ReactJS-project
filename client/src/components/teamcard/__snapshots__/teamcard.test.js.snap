// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamCard component should render TeamCard component 1`] = `ShallowWrapper {}`;
